src/main/java/de/cas/mse/exercise/SumDigPowerCalculator.java

```diff
@@ -0,0 +1,41 @@
+package de.cas.mse.exercise;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SumDigPowerCalculator {
+
+	public List<Long> calculate(long lowerBoundary, long upperBoundary) {
+		List<Long> result = new ArrayList<Long>();
+
+		for (long index = lowerBoundary; index < upperBoundary; index++) {
+			final String indexAsString = Long.toString(index);
+			final List<Long> splittedValues = split(indexAsString);
+			final Long powerSum = sumUpPowersOf(splittedValues);
+
+			final boolean powerSumEqualsIndex = powerSum == index;
+			if (powerSumEqualsIndex) {
+				result.add(powerSum);
```

Comment:
Variable name is longer than the expression and probably harder to understand, we think inlining would have been fine